src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1743,6 +1743,13 @@ public Task<(long Index, object Result)> SendToLeaderAsync(CommandBase cmd)
             return SendToLeaderAsyncInternal(cmd);
         }
 
+        //this is needed for cases where Result or any of its fields are blittable json.
+        //(for example, this is needed for use with AddOrUpdateCompareExchangeCommand, since it returns BlittableJsonReaderObject as result)
+        public Task<(long Index, object Result)> SendToLeaderAsync(TransactionOperationContext context, CommandBase cmd)
```

Comment:
I would rather _always_ use this. Is there a reason not to?

Reply:
The method with *public Task<(long Index, object Result)> SendToLeaderAsync(CommandBase cmd)* signature is used in gazillion of places.
Haven't checked ALL of them, but in some there are duplicate contexts used, and I guess SendToLeaderAsync() with context parameter CAN be used.

Reply:
We can do it now or in 4.2. After 4.1 RTM this change would not be a good option.

Maybe it would be better for the method that does not have an external context to do (in debug):
```
#if DEBUG
if (result.Result is BlittableJsonReaderBase)
    throw new InvalidOperationException();
#endif
```

Reply:
The problem isn't with direct dependency. It is if it returns _anything_ that holds BJRO

Reply:
And the problem with it being used everywhere it what scared me. Doing a deep check is fine with me on debug.

Reply:
Great, wanted to propose a deep check.

Reply:
In case of those commands, it returns a tuple where one of tuple's values is blittable

Reply:
http://issues.hibernatingrhinos.com/issue/RavenDB-11761